lib/baseFactory.js

```diff
@@ -55,13 +55,21 @@ class BaseFactory {
             table: this.table,
             params: {
                 id,
-                data: config
+                data: {}
             }
         };
 
+        // Filter for valid keys
+        this.model.allKeys.forEach((key) => {
+            if (config[key]) {
+                modelConfig.params.data[key] = config[key];
+            }
+        });
+
         return nodeify.withContext(this.datastore, 'save', [modelConfig])
-            .then(modelData => this.createClass(hoek.applyToDefaults(modelData, {
-                datastore: this.datastore
+            .then(modelData => this.createClass(hoek.applyToDefaults(config, {
+                datastore: this.datastore,
+                id: modelData.id
```

Comment:
The factories ask for it:

* [build](https://github.com/screwdriver-cd/models/blob/master/lib/buildFactory.js)
* [job](https://github.com/screwdriver-cd/models/blob/master/lib/jobFactory.js#L23)
* [pipeline](https://github.com/screwdriver-cd/models/blob/master/lib/pipelineFactory.js#L24)